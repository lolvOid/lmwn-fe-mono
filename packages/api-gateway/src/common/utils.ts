/**
 * A function that wraps the input function to handle asynchronous operations and errors.
 *
 * @param {Function} func - the input function to be wrapped
 * @return {Function} a function that handles asynchronous operations and errors
 */
export function asyncHandler(func: Function) {
    return (req: any, res: any, next: any) => Promise.resolve(func(req, res, next)).catch(next);
}