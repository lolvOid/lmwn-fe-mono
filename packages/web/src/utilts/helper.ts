import { SyntheticEvent, Dispatch, SetStateAction } from 'react';

export const getDiscountedPrice = (price: number | string | 0, discount: number | string | 0) => {
    const fullPrice = parseFloat(String(price)) || 0;
    return fullPrice - (fullPrice * parseFloat(String(discount))) / 100;
};

export const handleScroll = (
    event: SyntheticEvent,
    setScrolled: Dispatch<SetStateAction<boolean>>,
) => {
    const currentTarget = event.currentTarget;
    const offset = currentTarget.scrollTop;
    const containerHeight = currentTarget.scrollHeight - currentTarget.clientHeight - 1;

    if (offset >= containerHeight) {
        setScrolled(true);
    } else {
        setScrolled(false);
    }
};
