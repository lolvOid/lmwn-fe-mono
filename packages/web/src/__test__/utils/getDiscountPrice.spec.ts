import { getDiscountedPrice } from '@/utils/helper';

describe('getDiscountedPrice', () => {
  it('Should calculate discounted price when both price and discount are positive numbers', () => {
    const discountedPrice = getDiscountedPrice(100, 20);
    expect(discountedPrice).toMatchSnapshot();
  });

  it('Should handle string input for price and discount and calculate discounted price', () => {
    const discountedPrice = getDiscountedPrice('150', '25');
    expect(discountedPrice).toMatchSnapshot();
  });

  it('Should handle zero price and calculate discounted price', () => {
    const discountedPrice = getDiscountedPrice(0, 15);
    expect(discountedPrice).toMatchSnapshot();
  });

  it('Should handle zero discount and return the full price', () => {
    const discountedPrice = getDiscountedPrice(50, 0);
    expect(discountedPrice).toMatchSnapshot();
  });

  it('Should handle negative price and calculate discounted price', () => {
    const discountedPrice = getDiscountedPrice(-80, 10);
    expect(discountedPrice).toMatchSnapshot();
  });

  it('Should handle invalid input and return NaN', () => {
    const discountedPrice = getDiscountedPrice('invalid', 'invalid');
    expect(discountedPrice).toMatchSnapshot();
  });
});
