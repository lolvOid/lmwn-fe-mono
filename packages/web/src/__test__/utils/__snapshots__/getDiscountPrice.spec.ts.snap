// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDiscountedPrice Should calculate discounted price when both price and discount are positive numbers 1`] = `80`;

exports[`getDiscountedPrice Should handle invalid input and return NaN 1`] = `NaN`;

exports[`getDiscountedPrice Should handle negative price and calculate discounted price 1`] = `-72`;

exports[`getDiscountedPrice Should handle string input for price and discount and calculate discounted price 1`] = `112.5`;

exports[`getDiscountedPrice Should handle zero discount and return the full price 1`] = `50`;

exports[`getDiscountedPrice Should handle zero price and calculate discounted price 1`] = `0`;
